scripts/dumps2json.py

```diff
@@ -0,0 +1,176 @@
+                        assetsDict[assetName]['iam_policy'] = metadata['iam_policy']
+                if (i % 500 == 0) or (i == numberOfAssets):
+                    print(
+                        f"{i}/{numberOfAssets} caiDump: {dumpFilename}")
+
+    assetsDict = get_ancestry_path(assets=assetsDict)
+    assetsDict = remove_data(assets=assetsDict)
+
+    for _, asset in assetsDict.items():
+        assetsList.append(asset)
+        if asset['ancestry_path'] == "":
+            print(f"Warning: missing ancestry path {asset['name']}")
+
+    localJSONFile = open(outputFolderPath + '/' + outputFileName, 'w')
+    localJSONFile.write(json.dumps(assetsList, indent=2))
+    localJSONFile.close()
+
+
+def get_ancestry_path(assets):
+    # First flush, get the root asset
+    for _, asset in assets.items():
+        if asset['asset_type'] == "cloudresourcemanager.googleapis.com/Organization":
+            organizationPath = asset['resource']['data']['name'].replace(
+                'organizations', 'organization')
+            asset['ancestry_path'] = organizationPath
+
+    if organizationPath:
+        print(f"Found the organization for this dumps: {organizationPath}")
+        # Second flush, fix assets without resource object
+        for _, asset in assets.items():
+            if not 'resource' in asset:
+                asset['ancestry_path'] = organizationPath
+                print(
+                    f"warning: resource object is missing, ancestry_path set to org_path for {asset['name']}")
+
+        # Thrid flush, compute the ancestry path when possible
+        for _, asset in assets.items():
+            if not 'ancestry_path' in asset:
+                asset['ancestry_path'] = build_path(
+                    name=asset['name'], assets=assets)
+
+        knownProjects = {}
+        for _, asset in assets.items():
+            if 'ancestry_path' in asset and asset['asset_type'] == "cloudresourcemanager.googleapis.com/Project":
+                if not asset['resource']['data']['name'] in knownProjects:
+                    knownProjects[asset['resource']['data']
+                                  ['name']] = asset['ancestry_path']
+
+        # Fourth flush, fix it as possible
+        for _, asset in assets.items():
+            if asset['ancestry_path'] == "":
+                assetNameBreakdown = asset['name'].split('/')
+                if assetNameBreakdown[3] == 'projects':
+                    if assetNameBreakdown[4] in knownProjects:
+                        asset['ancestry_path'] = knownProjects[assetNameBreakdown[4]]
+                    else:
+                        asset['ancestry_path'] = organizationPath
+                elif assetNameBreakdown[3] == 'billingAccounts':
+                    asset['ancestry_path'] = 'organization/'
+                else:
+                    asset['ancestry_path'] = organizationPath
+                #print(f"warning fixed ancestry path for: {asset['name']}")
+    return assets
```

Comment:
We should be able to rewrite the method as below, in order to gain one loop, and use a dict comprehension to speed things up for the `knownProjects` var (needs testing):

```py
def get_ancestry_path(assets):

    # First flush, get the organization path
    organizationPath = None
    for _, asset in assets.items():
        if asset['asset_type'] == "cloudresourcemanager.googleapis.com/Organization":
            organizationPath = asset['resource']['data']['name'].replace(
                'organizations', 'organization')
            asset['ancestry_path'] = organizationPath

    if not organizationPath:
        print(f"Could not find the organization for this dumps")
        return assets

    # Second flush: add 'ancestry_path' field to all assets
    for _, asset in assets.items():

        # Fix assets without resource object
        if not 'resource' in asset:
            asset['ancestry_path'] = organizationPath
            print(
                f"warning: resource object is missing, ancestry_path set to org_path for {asset['name']}")

        # Compute the ancestry path if missing
        if not 'ancestry_path' in asset:
            asset['ancestry_path'] = build_path(
                name=asset['name'],
                assets=assets)

    # Third flush: Get map project name / ancestry for all projects
    knownProjects = {
        asset['resource']['data']['name']: asset['ancestry_path']
        for asset in assets
        if 'ancestry_path' in asset and
            asset['asset_type'] = "cloudresourcemanager.googleapis.com/Project"
    }

    # Fourth flush: loop through all assets and set 'ancestry_path' field if missing
    for _, asset in assets.items():
        if not asset['ancestry_path']:
            assetNameBreakdown = asset['name'].split('/')
            if assetNameBreakdown[3] == 'projects':
                if assetNameBreakdown[4] in knownProjects:
                    asset['ancestry_path'] = knownProjects[assetNameBreakdown[4]]
                else:
                    asset['ancestry_path'] = organizationPath
            elif assetNameBreakdown[3] == 'billingAccounts':
                asset['ancestry_path'] = 'organization/'
            else:
                asset['ancestry_path'] = organizationPath

    return assets
```

Reply:
TBD